src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/SourceGenerators/SourceGeneratorProjectItem.cs

```diff
@@ -47,7 +48,19 @@ public override Stream Read()
             => throw new NotSupportedException("This API should not be invoked. We should instead be relying on " +
                 "the RazorSourceDocument associated with this item instead.");
 
-        public bool Equals(SourceGeneratorProjectItem? other) => other is not null && AdditionalText == other.AdditionalText;
+        public bool Equals(SourceGeneratorProjectItem? other)
+        {
+            if (ReferenceEquals(AdditionalText, other?.AdditionalText))
+            {
+                return true;
+            }
+
+            // in the compiler server when the generator driver cache is enabled the
+            // additional files are always different instances even if their content is the same
+            var thisHash = AdditionalText.GetText()?.GetChecksum() ?? [];
+            var otherHash = other?.AdditionalText.GetText()?.GetChecksum() ?? [];
```

Comment:
@CyrusNajmabadi added a new method that is more correct when comparing the actual in-memory of two SourceText instances. Should this code be using [`SourceText.GetContentHash()`](https://github.com/dotnet/roslyn/blob/d9f63b6c1690af35225940ad16c3511361235240/src/Compilers/Core/Portable/Text/SourceText.cs#L620)?

Reply:
Oh good catch. I was using that, and switched over to checksum to benchmark to see what the difference was and forgot to put it back 🙃